modules/nf-core/find/concatenate/main.nf

```diff
@@ -0,0 +1,101 @@
+    def args2 = task.ext.args2 ?: ''
+    def file_list = files_in.collect { it.toString() }
+
+    // choose appropriate concatenation tool depending on input and output format
+
+    // | input     | output     | command1 | command2 |
+    // |-----------|------------|----------|----------|
+    // | gzipped   | gzipped    | cat      |          |
+    // | ungzipped | ungzipped  | cat      |          |
+    // | gzipped   | ungzipped  | zcat     |          |
+    // | ungzipped | gzipped    | cat      | pigz     |
+
+    // Use input file ending as default
+    prefix = task.ext.prefix ?: "${meta.id}${getFileSuffix(file_list[0])}"
+    file_extensions = getSetofFileSuffixes(file_list)
+    pattern_string = generatePatternString(file_extensions.toList())
+
+    out_zip = prefix.endsWith('.gz')
+    in_zip = file_list[0].endsWith('.gz')
+    command1 = in_zip && !out_zip ? 'zcat' : 'cat'
+    command2 = !in_zip && out_zip ? "| pigz -c -p ${task.cpus} ${args2}" : ''
+    if (file_list.contains(prefix.trim())) {
+        error(
+            "The name of the input file can't be the same as for the output prefix in the " + "module FIND_CONCATENATE (currently `${prefix}`). Please choose a different one."
+        )
+    }
+
+    """
+    find . -maxdepth 1 \\( -not -name '.*'  ${pattern_string} \\) \\
+         -exec sh -c "${command1} ${args} {} ${command2} >> ${prefix}" \\;
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        pigz: \$( pigz --version 2>&1 | sed 's/pigz //g' )
+    END_VERSIONS
+    """
+
+    stub:
+    def file_list = files_in.collect { it.toString() }
+    prefix = task.ext.prefix ?: "${meta.id}${file_list[0].substring(file_list[0].lastIndexOf('.'))}"
+    if (file_list.contains(prefix.trim())) {
+        error(
+            "The name of the input file can't be the same as for the output prefix in the " + "module FIND_CONCATENATE (currently `${prefix}`). Please choose a different one."
+        )
+    }
+    """
+    touch ${prefix}
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        pigz: \$( pigz --version 2>&1 | sed 's/pigz //g' )
+    END_VERSIONS
+    """
+}
+
+// for .gz files also include the second to last extension if it is present. E.g., .fasta.gz
+def getFileSuffix(filename) {
+    def match = filename =~ /^.*?((\.\w{1,5})?(\.\w{1,5}\.gz$))/
+    return match ? match[0][1] : filename.substring(filename.lastIndexOf('.'))
+}
+
+def getSetofFileSuffixes(in_files) {
+    return in_files.collect { getFileSuffix(it) }.toSet()
+}
+
+def generatePatternString(fileExtensionList) {
+    if (!fileExtensionList || fileExtensionList.isEmpty()) {
+        return ""
+    }
+
+    if (fileExtensionList.size() == 1) {
+        return "-name '*${fileExtensionList[0]}'"
+    }
+
+    def patternString = "-name '*${fileExtensionList[0]}' "
+    fileExtensionList[1..-1].each {
+        patternString += "-o -name '*${it}' "
+    }
+    return patternString.trim()
+}
```

Comment:
that seems slightly overcomplicated.

Are you aware of the Path getting attributes: https://www.nextflow.io/docs/latest/reference/stdlib.html#path

Reply:
Heh, great minds! But you gave more info.

To be fair , I think the getFileSuffix is copied from the original CAT_CAT module from @FriederikeHanssen

Reply:
It is but conditionally getting the extension preceding `.gz` is important for gzipped files